marimo/_ast/cell.py

```diff
@@ -289,6 +289,45 @@ def namespace_to_variable(self, namespace: str) -> Name | None:
     def is_coroutine(self) -> bool:
         return _is_coroutine(self.body) or _is_coroutine(self.last_expr)
 
+    @property
+    def is_toplevel_acceptable(self) -> bool:
```

Comment:
Is there a relationship to `_runtime.is_pure_function()`?

Reply:
Many similarities, but is_pure_function uses runtime information on variables, while everything here has to be top level and I think can be resolved during static analysis